### PR TITLE
fix(shub): apply -q to security hub

### LINKED ISSUE #1624: [Bug]: Prowler v3 integration to AWS Security Hub and usage of -q flag to send only FAILs
### Discussed in https://github.com/prowler-cloud/prowler/discussions/1623

<div type='discussions-op-text'>

<sup>Originally posted by **am-avios** December 28, 2022</sup>
Hello All

With previous version of prowler i was able to send only failed checks to security hub with flag -q, however it seems v3 sends both passed and failed checks to security hub even with -q flag. Is this intentional?  </div>

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Fixing https://github.com/prowler-cloud/prowler/issues/1624, apply -q option to Security Hub so only fails are sent.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/lib/security_hub/security_hub.py [hunk 1]
```diff
@@ -15,34 +15,42 @@
 
 
 def send_to_security_hub(
```
Comment (jfagoagas) [inline prowler/providers/aws/lib/security_hub/security_hub.py:17]: Would you mind to include a brief description using the docstring format?

```python
def send_to_security_hub(....):
"""
send_to_security_hub does the following and return X in the case of and Y in the case of...
"""
```

File: prowler/providers/aws/lib/security_hub/security_hub.py [hunk 2]
```diff
@@ -15,34 +15,42 @@
...
-            logger.error(
-                f"Security Hub is enabled in {region} but Prowler integration does not accept findings. More info: https://github.com/prowler-cloud/prowler/#security-hub-integration"
-            )
+            # Check if Prowler integration is enabled in Security Hub
+            if "prowler/prowler" not in str(
+                security_hub_client.list_enabled_products_for_import()
+            ):
+                logger.error(
+                    f"Security Hub is enabled in {region} but Prowler integration does not accept findings. More info: https://github.com/prowler-cloud/prowler/#security-hub-integration"
+                )
```
Comment (jfagoagas) [inline prowler/providers/aws/lib/security_hub/security_hub.py:42]: Why this if we are always sending findings to Security Hub?

File: prowler/providers/aws/lib/security_hub/security_hub.py [hunk 3]
```diff
@@ -15,34 +15,42 @@
...
             )
-
+            if batch_import["FailedCount"] > 0:
+                failed_import = batch_import["FailedFindings"][0]
+                logger.error(
+                    f"Failed to send archived findings to AWS Security Hub -- {failed_import['ErrorCode']} -- {failed_import['ErrorMessage']}"
+                )
+            success_count = batch_import["SuccessCount"]
     except Exception as error:
         logger.error(f"{error.__class__.__name__} -- {error} in region {region}")
```
Comment (jfagoagas) [inline prowler/providers/aws/lib/security_hub/security_hub.py]: Can you include the `traceback` to get the error line number? Thanks!

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/lib/security_hub/security_hub.py (hunk 2 above)
Comment (MrCloudSec) [reply prowler/providers/aws/lib/security_hub/security_hub.py:42]: We need to check if Prowler Integration is enabled.

File: prowler/providers/aws/lib/security_hub/security_hub.py (hunk 3 above)
Comment (MrCloudSec) [reply prowler/providers/aws/lib/security_hub/security_hub.py]: Sure!

File: prowler/providers/aws/lib/security_hub/security_hub.py (hunk 2 above)
Comment (jfagoagas) [reply prowler/providers/aws/lib/security_hub/security_hub.py:42]: Yes but if it is not enabled we are sending the findings anyway, is that correct?

Comment (MrCloudSec) [reply prowler/providers/aws/lib/security_hub/security_hub.py:42]: You are right! I corrected it.

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**